### PR TITLE
Use charm pkg updates

### PR DESCRIPTION
**Pull request by hmlanigan**
Updates related to pulling in the changes to the charm package. 

- Adding manifest to charm.
- Including implementation of a local ComputedSeries method, which now requires a manifest and metadata.
- Use charm version of Base, not systems.
- Remove use of systems pkg from juju.   <-- incomplete

## QA steps

Must use staging to qa.
Breaking no current functionality.  It's known there are issues with kubernetes charms from charmhub staging with bases today.  The `ambassador` charm fails in model.ValidateSeries.  `facundo-snappass-test` deploys as it has a manifest.

juju ssh to side card charm

### CODE REVIEW

File: apiserver/common/applicationwatcher.go [hunk 1]
```diff
@@ -157,6 +155,13 @@ func (w *applicationWatcher) handle(changes []string) ([]string, error) {
 	return filteredChanges, nil
 }
 
+func metadataFormatV2(manifest *charm.Manifest) charm.Format {
+	if manifest != nil && len(manifest.Bases) > 0 {
```
Comment (manadart) [inline apiserver/common/applicationwatcher.go:159]: @SimonRichardson and I were discussing how the V1 and V2 shouldn't be leaking out of the charm package.

File: state/charm.go [hunk 2]
```diff
@@ -737,7 +746,15 @@ func validateCharmSeries(modelType ModelType, series string, ch hasMeta) error {
...
-	return model.ValidateSeries(model.ModelType(modelType), series, ch.Meta().Format())
+
+	// TODO (manadart 2021-04-15): This is copied from ReadCharm in the charm
+	// package. It should be encapsulated there instead of repeated.
+	format := charm.FormatV2
+	if ch.Manifest() == nil || len(ch.Manifest().Bases) == 0 {
+		format = charm.FormatV1
+	}
+
+	return model.ValidateSeries(model.ModelType(modelType), series, format)
```
Comment (SimonRichardson) [inline state/charm.go:757]: This is actually wrong... we should re-write this and `model.ValidateSeries` in a follow up commit.

**Review by SimonRichardson — APPROVED** [review]

**Comment by manadart** [comment]
$$merge$$

**Comment by manadart** [comment]
$$merge$$

File: apiserver/common/applicationwatcher.go (hunk 1 above)
Comment (hmlanigan) [reply apiserver/common/applicationwatcher.go:159]: let's do a follow up

File: state/charm.go (hunk 2 above)
Comment (hmlanigan) [reply state/charm.go:757]: https://github.com/juju/juju/pull/12888

**Comment by hmlanigan** [comment]
$$merge$$

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**